react/features/remote-video-menu/components/AbstractKickButton.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import { openDialog } from '../../base/dialog';
+import { AbstractButton } from '../../base/toolbox';
+import type { AbstractButtonProps } from '../../base/toolbox';
+
+import { KickRemoteParticipantDialog } from '.';
+
+export type Props = AbstractButtonProps & {
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function,
+
+    /**
+     * The ID of the participant that this button is supposed to kick.
+     */
+    participantID: string,
+
+    /**
+     * The function to be used to translate i18n labels.
+     */
+    t: Function
+};
+
+/**
+ * An abstract remote video menu button which kicks the remote participant.
+ */
+export default class AbstractKickButton extends AbstractButton<Props, *> {
```

Comment:
Can we make a separate commit for this? This change was not part of my suggested improvements, it's a new feature if you will, and it's very self-contained.

Reply:
I don't think this is a new feature. This should be like this originally when I implemented it (separate abstract and platform dependant code), so I'd leave it here, if you don't mind

Reply:
Ok